src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp

```diff
@@ -59,15 +59,16 @@ class BiotPorosityUpdates : public PorosityBaseUpdates
     m_grainBulkModulus( grainBulkModulus )
   {}
 
+  GEOSX_HOST_DEVICE
+  real64 getBiotCoefficient( localIndex const k ) const { return m_biotCoefficient[k]; }
 
   GEOSX_HOST_DEVICE
   void updatePorosity( localIndex const k,
                        localIndex const q,
                        real64 const & deltaPressure,
                        real64 const ( &strainIncrement )[6],
                        real64 & dPorosity_dPressure,
-                       real64 & dPorosity_dVolStrain,
-                       real64 & dTotalStress_dPressure ) const
+                       real64 & dPorosity_dVolStrain ) const
```

Comment:
Since you are doing this can you rename this function `updatePorosityFromPressureAndStrain` and then also rename the one in `PressurePorosity`  `updatePorosityFromPressure`. This way they are more similar to each other.

Reply:
So to update porosity function will become:

- `updatePorosityFromPressure` instead of `updateFromPressure` in `PressurePorosity`
- `updatePorosityFromPressureAndStrain` instead of `updatePorosity` in `BiotPorosity`
- `updatePorosityFromProppantVolumeFraction` instead of `updateFromProppantVolumeFraction` in `ProppantPorosity`

How about just having `updatePorosity` everywhere?

Reply:
If I m not mistaking the compilers don't love having virtual methods with the same name and just a different signature. I am almost sure I was getting warnings.

Reply:
I'd be tempted to have a non-virtual method `update` defined in each derived class. But I know people may be opposed to this solution.